src/dyn/dyn_hop_proposal.cpp

```diff
@@ -1310,8 +1310,8 @@ int hop(vector<double>& prob, double ksi){
       } // for
 
     } // if nrm
-    else{  finstate = initstate; }  // probability to hop to any other states is zero
-                                    // so stay on the original state
+    else{  finstate = 0; }   // probability to hop to any other states is zero

```

Comment:
this is incorrect - in this case you just hop to the zeroth state

Reply:
This routine does not have initstate. Original state is not known. 

There is a three-arg variant with initstate in first argument and vector at second argument:
https://github.com/Quantum-Dynamics-Hub/libra-code/blob/14a99c738ef09d9a58ccdc83aeab611e72bbc2e1/src/dyn/dyn_hop_proposal.cpp#L1361-L1402

Probably, one needs to use this.